app/PhoneNumber.php

```diff
@@ -27,27 +30,40 @@ public function markVerified()
         $this->save();
     }
 
-    public static function findByNumber($number)
+    public function markBlacklisted($value = true)
     {
-        return self::where('number', $number)->firstOrFail();
+        $this->is_blacklisted = $value;
+        $this->save();
+        info('Number was blacklisted', ['class' => get_called_class(), 'id' => $this->id]);
     }
 
     public function scopeVerified(Builder $builder)
     {
         return $builder->where('is_verified', true);
     }
 
-    public static function findByTwilioNumber($number)
+    public function scopeBlacklisted(Builder $builder, $value = true)
```

Comment:
Are there any places in the site where we need to be default scoping to "not blacklisted" that we're not? Or do you have that all covered?